src/neural/network.h

```diff
@@ -106,7 +106,9 @@ class Network {
  public:
   virtual const NetworkCapabilities& GetCapabilities() const = 0;
   virtual std::unique_ptr<NetworkComputation> NewComputation() = 0;
+  virtual int GetThreads() const { return 1; }
```

Comment:
(optional)
I'd suggest to move GetThreads(), IsCpu() and GetMinibatchSize() into the Capabilities struct.
(or have a separate struct for them if you feel that they don't really belong to Capabilities -- although then better rename "Capabilities" to "Properties").
Otherwise it's kind of spoiling the interface -- it's clearly the information leakage, but at least it would be isolated.